### ISSUE TITLE
[Feature] Let vampires actively choose between biting/drinking/turning

### ISSUE THREAD

**Issue by lunofe** [TeamLapen/Vampirism#805]
In my opinion, vampires should be able to actively decide, whether to drink blood / bite an entity, or to turn it.

*(I suggested this on Discord a while ago, but it didn't quite work out. However, I've invested some time in refining the concept further.)*

---

#### In most modern stories/universes containing vampires, a transformation works like this:
1. The vampire drinks a little amount of blood from the human
2. The human needs to drink a bigger amount of blood from the vampire (most commonly the vampire bites his own wrist for this)
3. (Optional: The human must die, to be reborn as a vampire) *
4. The human transformed into a vampire and feels a rush of adrenaline and the extended senses for a few moments (equivalent to the saturation effect in MC?)

_* This would heavily change the way Vampirism does it currently and would lead to further questions, eg. "should the player loose their items on this particular death", etc., so I'd say this is **really** optional and can stay how it is currently (sanguinare effect)._

---

#### So, how could the transformation look like in Vampirism?
Currently there's one keybind to utilize the fangs. Pressing it will deal damage, holding it will (try) to consume blood. That's a pretty good solution. However, turning a human into a vampire should be different from just drinking their blood by holding the key.

The ability to turn someone could be an unlockable skill, rising the current maximum level from 14 to 15: 
![tree_concept](https://user-images.githubusercontent.com/42828197/98587310-3c93f600-22ca-11eb-8252-8a7fd79e2ab8.png)

Alternatively, if you don't want to make it it's own unlockable skill resulting in 15 levels, the ability could also be part of the current level 4, renaming it to "No longer a fledgling" and unlocking both bat mode and the ability to turn someone. *I really like the sound of that.* Players who didn't unlock the skill yet, should not be able to turn players, but randomly infect other entities, basically how it works currently. After unlocking the skill, they're no longer a fledgling and can actively control it now.

For the actual turning, I've been thinking of a (probably?) rather easy solution. Point at the entity you want to turn, open your action menu and select the turn skill.

---

#### About other entities...
Currently, mobs either die or turn once their blood is depleted. That should be changed as well:
- Pressing the (current) key: deal damage
- Holding the (current) key: drink blood
- Using the new skill: turn the entity with 100% success rate

If an entity suffers from blood loss (aka. is drained dry), it should just die, without having the chance to turn, just as players do.

---

#### To sum up...
- The current key press/hold mechanic should definitely stay the same:
     - Pressing: deal damage
     - Holding: drink blood
- The new skill to turn entities should be accessible from the action menu and should support being bound to a key using ``/vampirism bind-action``
- Vampire NPCs, the fang item & a sanguinare syringe should still be able to turn a player like they do now
- Entities should follow the same rules as players, when being the subject to being bitten (damage/blood loss/turn, depends on what the vampire has chosen)

**Comment by maxanier** [comment]
Very well conceived and described.

I don't think we should require a skill point for this. It is only relevant for some people, which either play on a certain multiplayer server or are very much interested in the lore. Everybody else should not have to spend an additional skill point on this.
So I like the "No longer a fledgling" idea.

Everything else sounds good. We just have to make sure to communicate this to players appropriately, so we don't get another disaster like when holding the key became necessary for feeding.

**Comment by Dover2** [comment]
a simple fix (at least i think its simple conceptually) would be to have level 4 or level 5 grant 2 skill points, to accommodate for this extra upgrade, so that you wont have to make it level 15.

another method would be to make this upgrade a free one or just to include it with the bat transformation, similar to how OP put it.

**Comment by lunofe** [comment]
Now that I think of it, adding yet another level just for this skill might not only be a little overkill since it's just useful for multiplayer, but also lead to confusion, since there has been a secret level 15 in 1.12.2. So yeah, the rebranding of level 4 to "no longer a fledgling" definitely seems like the best idea.

**Comment by maxanier** [comment]
Should the turning happen right upon infection or should there also be a short (maybe 30s) sanguinare effect before the infected entity is actually turned (like with the old "realism mode")?

**Comment by lunofe** [comment]
Well, I would say that the delay with the Sanguinare effect stays the same for players, meaning it's one day. For other mobs, though, that would be a bit annoying, so I would suggest maybe 30 seconds or less - just not completely without a delay, that would be kind of strange in my opinion and would hurt the flair of it.
